hawkular-bus/hawkular-bus-common/src/main/java/org/hawkular/bus/common/BinaryData.java

```diff
@@ -59,68 +59,52 @@ public void setOnCloseAction(Runnable action) {
         onCloseAction = action;
     }
 
-    public int read() throws IOException {
-        if (unreadInMemoryDataExists()) {
-            return (inMemoryData[inMemoryDataPointer++] & 0xff);
-        } else {
-            return streamData.read();
-        }
-    }
+    public void close() throws IOException {
+        backingStream.close();
```

Comment:
I don't think we should. We just delegate the calls, so, either the caller catches it, or the backing stream handles what is to be handled.